homework/index.js

```diff
@@ -28,26 +30,133 @@
     });
     return elem;
   }
+  function formatDate(dateString) {
+    const dateTime = new Date(dateString);
+    return dateTime.toLocaleString();
+  }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function renderError(error) {
+    const container = document.getElementById('main-container');
+    container.innerHTML = '';
+    createAndAppend('div', container, {
+      text: error.message,
+      class: 'alert',
+    });
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, {
-          text: err.message,
-          class: 'alert-error',
+  function addTable(table, body, value) {
+    const tr = createAndAppend('tr', table);
+    createAndAppend('th', tr, { text: body });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepoDetails(repo) {
+    const leftSide = document.getElementById('left-side');
+    leftSide.innerHTML = '';
+    const ul = createAndAppend('ul', leftSide);
+    const li = createAndAppend('li', ul);
+    const table = createAndAppend('table', li);
+    const tr1 = addTable(table, 'Repository:', '');
+    createAndAppend('a', tr1.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+    });
+    addTable(table, 'Description:', repo.description);
+    addTable(table, 'Fork: ', repo.forks);
+    addTable(table, 'Updated:', formatDate(repo.updated_at));
+  }
+
+  function fetchAndRender(repo) {
+    const rightSide = document.getElementById('right-side');
+    rightSide.innerHTML = '';
+    fetchJSON(repo.contributors_url)
+      .then(contributors => {
+        createAndAppend('span', rightSide, {
+          text: 'Contributions',
+          class: 'title',
         });
-        return;
-      }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+        const contributorList = createAndAppend('ul', rightSide, {
+          class: 'contributor-list',
+        });
+        contributors.forEach(contributor => {
+          const list = createAndAppend('li', contributorList, {
+            class: 'contributor-item',
+          });
+          createAndAppend('img', list, {
+            class: 'image',
+            src: contributor.avatar_url,
+          });
+          createAndAppend('span', list, {
+            class: 'contributor-name',
+            text: contributor.login,
+          });
+          createAndAppend('span', list, {
+            class: 'contribution-count',
+            text: contributor.contributions,
+          });
+          list.addEventListener('click', () => {
+            window.open(contributor.html_url, '_blank');
+          });
+          list.addEventListener('key', event => {
+            if (event.key === 'Enter')
+              window.open(contributor.html_url, '_blank');
+          });
+        });
```

Comment:
This `fetchAndRender()` function is rather large. For readability, I would advise to split this function by creating a separate function that contains all the code that you now have in the arrow function in lines 82-105. This function needs two parameters:

```js
function renderContributor(contributor, contributorList) {
  const list = createAndAppend('li', contributorList, {
    class: 'contributor-item',
  });
  ...
}
```

Then, in line 82 you can call this function like this:

```js
contributors.forEach(contributor =>
  renderContributor(contributor, contributorList),
);
```